client/src/main/java/org/eclipse/hono/client/CommandContext.java

```diff
@@ -106,8 +106,10 @@ public ProtonDelivery getDelivery() {
     public void accept() {
         final Span span = getTracingSpan();
         LOG.trace("accepting command message [{}]", getCommand());
-        ProtonHelper.accepted(delivery, true);
-        span.log("accepted command for device");
+        if (checkDeliveryNotAlreadySettled(DeliveryState.DeliveryStateType.Accepted)) {
```

Comment:
why shouldn't we settle the message locally if the peer has sent an (unexpected) disposition frame? IMHO we should always settle in order to let the AMQP protocol stack to do its housekeeping, e.g. removing the transfer from the *unsettled* list.

Reply:
I've refactored the CommandContext classes now, making sure the delivery gets settled locally.